fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.util
+
+import spock.lang.Specification
+
+import java.nio.file.Files
+import java.nio.file.Paths
+
+class UtilsSpec extends Specification {
+
+    private static final TEMP_DIR = "target/tmp/util/"
+
+    def cleanup() {
+        Files.deleteIfExists(Paths.get(TEMP_DIR))
+    }
+
+    def "Get by sub-type returns all instances of that sub-type"() {
+        given: "a collection of sets which share the same HashSet class"
+        Collection set = [["a"] as HashSet, ["b"] as LinkedHashSet]
+
+        expect: "getting by LinkedHashSet returns LinkedHashSet instance"
+        Utils.getSubsetByType(set, LinkedHashSet) == [["b"] as LinkedHashSet] as LinkedHashSet
+
+        and: "getting by HashSet returns both HashSet and LinkedHashSet instances"
+        Utils.getSubsetByType(set, HashSet) == set as LinkedHashSet
+    }
+
+    def "File path without parent dir generates the parent"() {
+        given: "a file path without a parent directory"
+        String path = TEMP_DIR + "data.txt"
+
+        // make sure parent directory does not exist yet
+        assert ! new File(TEMP_DIR).exists()
```

Comment:
(minor stylistic) You can make this an `expect` block and and turn the comment into a block comment so you don't need the `assert`

Reply:
I like that. Thanks!